wordpress-mobile/WordPress-Android
# [Gutenberg] Set consistent editor colours in both the WordPress and Jetpack apps

Fixes the Android side of https://github.com/wordpress-mobile/gutenberg-mobile/issues/5047, [iOS PR is here](https://github.com/wordpress-mobile/WordPress-iOS/pull/19113)

## Description

In the Jetpack app, some UI elements like the cursor, text selection, and caret are using a green colour. However, the rest of the editor’s elements are using blue. 

To be consistent with colours across the editor, this PR seeks to update them for consistency across both the WordPress and Jetpack apps. The PR specifically achieves the following for the Jetpack app:

* Updates the colour of UI elements within the main editor (e.g. cursor, highlight) from green to blue.
* Updates the accent of sub screens within Post Settings from green to blue.

There's still some known inconsistencies that need to be addressed in separate PRs, however:

* The media picker (available from media blocks and the Post Settings screen) still has some green.
* There is still green visible in other screens 'connected' to the editor, such as Preview, History, Help & Support, etc. 

## Screenshots

A before/after screenshot taken from the Jetpack app to show the editor's green → blue colour changes:

| Before  | After |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/182917626-a3c35296-f1ca-40e5-afa0-0974cb350f6e.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/182917679-a7aab2e0-5d34-4a63-a1b2-a2ff7cf0f83c.png" width="100%"> |

## Testing

* Download this PR's installable build for both the WordPress and Jetpack apps.
* In each app, navigate to the editor.
* Verify the colours used throughout the editor appear consistent for both app. The cursor, text selection, and caret should all now be blue for Jetpack (this is in contrast to the green that's used outside of the app's editor). There should be no change in the WordPress app, with the colours remaining blue.  

Special considerations to keep in mind while testing include:

* This PR aims to bring colour consistency to the Post Settings screen as well as the editor.
* Testing of any popovers, modals, or less visible UI elements within the editor would be beneficial. 
* As this is a UI change, dark/light mode should be a consideration when testing.
* These changes should be noticeable in both the post _and_ the page editor.
* There should be no changes to colours outside of the editor. Screens that may be used in more one place, such as category selection, should be tested.

## Code Changes

### React Native RootView

* A new theme, `GutenbergContainerTheme` was [created under the editor's `styles.xml` file](https://github.com/wordpress-mobile/WordPress-Android/blob/91a25da6e13dfd06e7b6bf105fa00e13c2468d44/libs/editor/src/main/res/values/styles.xml#L7-L9). This theme references [an accent colour that equals the same hex code](https://github.com/wordpress-mobile/WordPress-Android/blob/91a25da6e13dfd06e7b6bf105fa00e13c2468d44/libs/editor/src/main/res/values/colors.xml#L33) as the WordPress app's primary colour. This new theme is then applied to [the editor's fragment.xml](https://github.com/wordpress-mobile/WordPress-Android/blob/91a25da6e13dfd06e7b6bf105fa00e13c2468d44/libs/editor/src/main/res/layout/fragment_gutenberg_editor.xml#L4).

### Post Settings 

* A new `colorPrimaryEditor` variable has been added to `colors_base.xml` for both WordPress ([light mode](https://github.com/wordpress-mobile/WordPress-Android/blob/44882b38bf23e65c59a63ce0fde7d06cefac04a2/WordPress/src/main/res/values/colors_base.xml#L8) / [dark mode](https://github.com/wordpress-mobile/WordPress-Android/blob/44882b38bf23e65c59a63ce0fde7d06cefac04a2/WordPress/src/main/res/values-night/colors_base.xml#L8)) and Jetpack ([light mode](https://github.com/wordpress-mobile/WordPress-Android/blob/44882b38bf23e65c59a63ce0fde7d06cefac04a2/WordPress/src/jetpack/res/values/colors_base.xml#L8) / [dark mode](https://github.com/wordpress-mobile/WordPress-Android/blob/44882b38bf23e65c59a63ce0fde7d06cefac04a2/WordPress/src/jetpack/res/values-night/colors_base.xml#L8)).
* This new variable has been used to create a special `PostSettingsTheme` in the main app's `style.xml` ([light mode](https://github.com/wordpress-mobile/WordPress-Android/blob/8c4db328308a85e77fe6331d2bebc0a0fa0cdbb8/WordPress/src/main/res/values-night/styles.xml#L129-L135) / [dark mode](https://github.com/wordpress-mobile/WordPress-Android/blob/8c4db328308a85e77fe6331d2bebc0a0fa0cdbb8/WordPress/src/main/res/values-night/styles.xml#L129-L135)). 
* `PostSettingsTheme` can now be applied to any areas in the Post Settings screen that should remain blue across both the WordPress and Jetpack apps. For example, the [input dialogs](https://github.com/wordpress-mobile/WordPress-Android/blob/8c4db328308a85e77fe6331d2bebc0a0fa0cdbb8/WordPress/src/main/java/org/wordpress/android/ui/posts/PostSettingsInputDialogFragment.java#L87-L88) and [the add categories screen](https://github.com/wordpress-mobile/WordPress-Android/blob/8c4db328308a85e77fe6331d2bebc0a0fa0cdbb8/WordPress/src/main/res/layout/add_category.xml#L5).

## Regression Notes
1. Potential unintended areas of impact

There's the potential for this PR to unexpectedly change colours in either the WordPress or Jetpack app beyond what's desired. For example, the cursor colour may unexpectedly change within the WordPress app. 

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I manually tested both apps, making sure to experiment with various blocks and writing flows.

3. What automated tests I added (or what prevented me from doing so)

No automated tests.

<hr />

PR submission checklist:
